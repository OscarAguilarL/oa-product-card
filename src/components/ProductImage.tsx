import React, { useContext } from 'react';

import { productContext } from './ProductCard';

import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

export interface Props {
    className?: string;
    img?: string;
    style?: React.CSSProperties;
    alt?: string;
}

export const ProductImage = ({ img, className, style, alt }: Props) => {
    const { product } = useContext(productContext);
    let imgToShow: string;

    if (img) {
        imgToShow = img;
    } else if (product.img) {
        imgToShow = product.img;
    } else {
        imgToShow = noImage;
    }

    return (
        <img
            src={imgToShow}
            className={`${styles.productImg} ${className}`}
            alt={product ? product.title : alt}
            style={style}
        />
    );
};
