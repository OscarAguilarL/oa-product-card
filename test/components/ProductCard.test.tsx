import React from 'react';
import renderer from 'react-test-renderer';

import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';

const { act } = renderer;

describe('Pruebas en el componente ProductCard', () => {
  test('Debe de mostrar el componente correctamente', () => {
    const wrapper = renderer.create(
      <>
        <ProductCard product={product1}>
          {() => <h2>Product Card</h2>}
        </ProductCard>
      </>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de incrementar el contador', () => {
    const wrapper = renderer.create(
      <>
        <ProductCard product={product1}>
          {({ counter, increaseBy }) => (
            <>
              <h2>ProductCard</h2>
              <span>{counter}</span>
              <button onClick={() => increaseBy(1)}></button>
            </>
          )}
        </ProductCard>
      </>
    );

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
