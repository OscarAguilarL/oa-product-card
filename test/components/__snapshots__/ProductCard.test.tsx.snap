// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ProductCard Debe de incrementar el contador 1`] = `
<section
  className="productCard undefined"
>
  <h2>
    ProductCard
  </h2>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</section>
`;

exports[`Pruebas en el componente ProductCard Debe de mostrar el componente correctamente 1`] = `
<section
  className="productCard undefined"
>
  <h2>
    Product Card
  </h2>
</section>
`;
